kratos/processes/check_skin_process.h

```diff
@@ -90,34 +74,31 @@ class CheckSkinProcess: public Process
         Execute();
     }
 
-
     ///@}
     ///@name Operations
     ///@{
 
 
-    /// Check elements to make sure that their jacobian is positive and conditions to ensure that their face normals point outwards
+    /**
+     * @brief Check elements to make sure that their jacobian is positive and conditions to ensure that their face normals point outwards
+     */
     void Execute() override
     {
         KRATOS_TRY;
+            
+        KRATOS_ERROR_IF(mrModelPart.Conditions().size() == 0 && mrModelPart.Elements().size() != 0) << "The number of conditions is zero and the number of elements is not, hence the skin can not envelope the domain" << std::endl;
 
-        if(mrModelPart.Conditions().size() == 0 && mrModelPart.Elements().size() != 0)
-            KRATOS_THROW_ERROR(std::invalid_argument, "the number of conditions is zero and the number of elements is not, hence the skin can not envelope the domain","")
-
-        typedef std::unordered_map<DenseVector<unsigned int>, unsigned int, KeyHasherRange<DenseVector<unsigned int>>, KeyComparorRange<DenseVector<unsigned int>> > hashmap;
+        typedef std::unordered_map<DenseVector<IndexType>, IndexType, KeyHasherRange<DenseVector<IndexType>>, KeyComparorRange<DenseVector<IndexType>> > hashmap;
         hashmap edge_map;
 
-        DenseVector<unsigned int> ids(2);
+        DenseVector<IndexType> ids(2);
 
-        //add 1 to the counter for every edge find in the model part
-        for (ModelPart::ConditionIterator itCond = mrModelPart.ConditionsBegin(); itCond != mrModelPart.ConditionsEnd(); itCond++)
-        {
-            Element::GeometryType::GeometriesArrayType edges = itCond->GetGeometry().Edges();
+        // Add 1 to the counter for every edge find in the model part
+        for (auto& r_cond : mrModelPart.Conditions()) {
+            const auto edges = r_cond.GetGeometry().GenerateEdges();
```

Comment:
for the record this is not cosmetic

@rubenzorrilla I hope this is also ok for you